File: 58. Length of Last Word/lengthOfLastWord.js
/**
 * @param {string} s
 * @return {number}
 */
const lengthOfLastWord = (s) => {
  let i = s.length - 1;
  let lastWordLength = 0;

  while (s[i] === ' ') {
    i--;
  }

  while (i >= 0 && s[i] !== ' ') {
    lastWordLength++;
    i--;
  }

  return lastWordLength;
};

const s = "   fly me   to   the moon  ";
console.log('lengthOfLastWord: ', lengthOfLastWord(s));

File: 58. Length of Last Word/lengthOfLastWord.py
class Solution:
  def lengthOfLastWord(self, s: str) -> int:
    i, lastWordLength = len(s) - 1, 0
    
    while s[i] == ' ':
      i -= 1
      
    while i >= 0 and s[i] != ' ':
      lastWordLength += 1
      i -= 1
      
    return lastWordLength
  
s = "   fly me   to   the moon  "
solution = Solution()
print('lengthOfLastWord: ', solution.lengthOfLastWord(s))
